### DISCUSSION TITLE
How upstream connection speed impacts the result?

### DISCUSSION THREAD

**Discussion by nnn406** [esnet/iperf#1246] [Q&A] [answered]
Hi,

I'm running iperf3 server on my VPS

`ìperf3 -s`

from my home, I'm running the client like that

`iperf3 -c x.x.x.x -t30`

However, I can't get more than my upstream connection speed.
I've tried multiple times from my friends and they have a similar result. The value is different compare to me but it's everytime limited to their outgoing connection speed.

Is it expected behavior or not?

P.S. If I'm running speedtest from okla with the target server inside the country where my ISP is I see different result.
P.P.S If I'm running a speed test from https://github.com/librespeed/speedtest hosted on my VPS I also see better result

**Comment by bmah888** [comment, marked as answer]
By default, iperf3 sends test data from the client to the server...in your case it's from your home to the VPS. So getting (approximately at least) the upstream connection speed is expected. If you want to test in the opposite direction, you can try passing the `--reverse` flag to the iperf3 client, which would cause test data to be sent from the server to the client. However, please be aware that many home "routers" do network address translation (NAT), at least for IPv4, which iperf3 doesn't make any attempts to handle. So it might not actually work.

**Reply by nnn406** [reply]
could you please elaborate @bmah888 a bit about the NAT issue? Why iperf might need to handle it? I thought it's a router's duty to take care of that.

**Reply by bmah888** [reply]
Actually, disregard that part of my comment. It should work fine. In general it's difficult to establish TCP connections from the outside of a NAT to the inside (in typical residential broadband environments), but that doesn't really apply to this scenario.
